spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -145,6 +146,17 @@ public void setRequestChannelName(String requestChannelName) {
 	}
 
 	/**
+	 * Return the name of the request channel
+	 * @return the channel name.
+	 * @since 4.3
+	 */
+	public String getRequestChannelName() {
+		return this.requestChannel == null ? this.requestChannelName :
+			this.requestChannel instanceof NamedComponent ? ((NamedComponent) this.requestChannel).getComponentName()
+					: null;
```

Comment:
Yes. Agreed. The `MessagingGatewaySupport` may be used only for the `receive()`. Therefore no any `requestChannel`.
Fixed.
Never mind